torch/optim/optimizer.py

```diff
@@ -255,9 +255,14 @@ def __init__(self, params: ParamsT, defaults: Dict[str, Any]) -> None:
         self._patch_step_function()
 
         if isinstance(params, torch.Tensor):
-            raise TypeError("params argument given to the optimizer should be "
-                            "an iterable of Tensors or dicts, but got " +
-                            torch.typename(params))
+            if self.__class__.__name__ == 'SparseAdam':
```

Comment:
happy to use a different method than looking into the dunders here, just didn't know of another way